dali/pipeline/operators/decoder/nvjpeg_decoder.h

```diff
@@ -161,25 +162,33 @@ class nvJPEGDecoder : public Operator<MixedBackend> {
       EncodedImageInfo info;
 
       // Get necessary image information
-      NVJPEG_CALL(nvjpegGetImageInfo(handle_,
+      nvjpegStatus_t ret = nvjpegGetImageInfo(handle_,
                                      static_cast<const unsigned char*>(data), in_size,
                                      &info.c, &info.subsampling,
-                                     info.widths, info.heights));
+                                     info.widths, info.heights);
+      // Fallback for png
+      if (ret == NVJPEG_STATUS_BAD_JPEG && CheckIsPNG(static_cast<const uint8*>(data), in_size)) {
```

Comment:
What if it's not a JPEG but also not a PNG?  Is there something more general we could do here?

Reply:
Not really unfortunately (besides ripping something from ImageMagick or something like that) - basically codes like OpenCV do not advertise functionality for getting info from image header without decoding the image too.

Reply:
There is stuff like this: https://github.com/scardine/image_size the logic from which we could use to expand the capabilities here (which would be much easier than searching for that functionality in OpenCV/ImageMagick source)